# EDA 

In [2]:
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
import sqlite3 as sql

## 1. Cargando el dataset

In [3]:
ruta_archivo = "../data/raw/adidas_us_sales_dataset.xlsx"

df = pd.read_excel(ruta_archivo, index_col=None)

# Eliminar columna de índice anterior si existe
if 'Unnamed: 0' in df.columns:
    df.drop(columns=['Unnamed: 0'], inplace=True)

# Guardar en SQLite
conn = sql.connect("../data/raw/adidas_sales.db")
df.to_sql("ventas_adidas", conn, if_exists="replace", index=False)

print("Base de datos creada con éxito")

cursor = conn.cursor()


Base de datos creada con éxito


## 2. Conociendo la estructura del dataset

In [4]:
# Número de filas y columnas:
print(f"Dimensiones del dataset: {df.shape}")

# Información sobre los tipos de datos
print("\nInformación del dataset:")
df.info()

Dimensiones del dataset: (9648, 13)

Información del dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9648 entries, 0 to 9647
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Retailer          9648 non-null   object        
 1   Retailer ID       9648 non-null   int64         
 2   Invoice Date      9648 non-null   datetime64[ns]
 3   Region            9648 non-null   object        
 4   State             9648 non-null   object        
 5   City              9648 non-null   object        
 6   Product           9648 non-null   object        
 7   Price per Unit    9648 non-null   float64       
 8   Units Sold        9648 non-null   int64         
 9   Total Sales       9648 non-null   float64       
 10  Operating Profit  9648 non-null   float64       
 11  Operating Margin  9648 non-null   float64       
 12  Sales Method      9648 non-null   object        
dtypes: datetime64[ns

### Análisis:

- Tenemos 9648 filas y 13 columnas.
- Hay variables numéricas (float64, int64) y categóricas(object).
- No hay valores nulos en el dataset.

## 3. Revisando los nombres de las columnas

In [6]:
# Nombres de las columnas
print(f"Nombres de las columnas:")
df.columns.tolist()

Nombres de las columnas:


['Retailer',
 'Retailer ID',
 'Invoice Date',
 'Region',
 'State',
 'City',
 'Product',
 'Price per Unit',
 'Units Sold',
 'Total Sales',
 'Operating Profit',
 'Operating Margin',
 'Sales Method']

## 4. Tipos de variables detectadas

In [7]:
print("Tipos de variables en el dataset:")
print(df.dtypes)

Tipos de variables en el dataset:
Retailer                    object
Retailer ID                  int64
Invoice Date        datetime64[ns]
Region                      object
State                       object
City                        object
Product                     object
Price per Unit             float64
Units Sold                   int64
Total Sales                float64
Operating Profit           float64
Operating Margin           float64
Sales Method                object
dtype: object


### Interpretación:

- Numéricas: `Retailer ID`, `Price per Unit`, `Units Sold`, `Total Sales`, `Operating Profit`, `Operating Margin`.
- Categóricas: `Retailer`, `Region`, `State`, `City`, `Product`, `Sales Method`.
- Fechas: `Invoice Date`.